### ISSUE TITLE
No graphical session after update to 41.20241117.3 on Aurora.

### ISSUE THREAD

**Issue by mejiasjrg** [ublue-os/aurora#8]
### Describe the bug

The system updated from 41.20241112.1  to 41.20241117.3 and on the next boot it showed the arrow cursor for a little while and then a blank screen. It didn't even show the login screen. I changed to a terminal with _Ctrl-Alt-F2_ and could login with no issues. Tried _sudo systemctl restart sddm_, and got the blank screen again.
Rebooted the machine and selected the second entry in the grub menu, and the system booted normally to the graphical session. Then I issued _rpm-ostree rollback_, reboted, and issued _systemctl start rpm-ostreed-automatic.service_ to make the system update again, hoping there might be a new (different) update, but instead updated to 41.20241117.3 again, and again got the blank screen.
I finally did the rollback again and am posting from the system running 41.20241112.1

### What did you expect to happen?

I expected the system to boot normally running 41.20241117.3

### Output of `bootc status`

```shell
Current staged image: ghcr.io/ublue-os/aurora:stable
    Image version: 41.20241117.3 (2024-11-17 15:47:30 UTC)
    Image digest: sha256:fb3a023ce3c7609591c7524cb0c5a679453e5e0ff62cef6f334b4d7ca165965b
Current booted image: ghcr.io/ublue-os/aurora:stable
    Image version: 41.20241112.1 (2024-11-12 21:06:27 UTC)
    Image digest: sha256:f99e7cf789398600ffe7d1ade44ca0bceb31d76cbe8b8eb140a8b284804d6f6c
Current rollback image: ghcr.io/ublue-os/aurora:stable
    Image version: 41.20241117.3 (2024-11-17 15:47:30 UTC)
    Image digest: sha256:fb3a023ce3c7609591c7524cb0c5a679453e5e0ff62cef6f334b4d7ca165965b
```

### Output of `groups`

```shell
jm wheel
```

### Extra information or context

For the average user Aurora is intended to, this would have been a show-stopper. Not good.

**Comment by mejiasjrg** [comment]
I've been fiddling with `aurora:stable`, but as I've had not much time to do it, this spans from Dec-12 to this day. Anyway, here is my report.

The system currently running:
❯ rpm-ostree status
State: idle
AutomaticUpdates: stage; rpm-ostreed-automatic.timer: no runs since boot
Deployments:
● ostree-image-signed:docker://ghcr.io/ublue-os/aurora:stable
                   Digest: sha256:a5abc38a6c2f83546f8c4df05de254a43a8410fd6d2fff8fcf3d130abcc35df6
                  Version: 41.20241208.4 (2024-12-08T16:49:36Z)
          LayeredPackages: gdm lightdm lightdm-greeter lightdm-settings plasma-workspace-x11

  ostree-image-signed:docker://ghcr.io/ublue-os/aurora:stable
                   Digest: sha256:a5abc38a6c2f83546f8c4df05de254a43a8410fd6d2fff8fcf3d130abcc35df6
                  Version: 41.20241208.4 (2024-12-08T16:49:36Z)
          LayeredPackages: gdm lightdm lightdm-greeter lightdm-settings plasma-workspace-x11

Starting from `aurora:40` I installed `rpm-ostree install gdm lightdm lightdm-greeter lightdm-settings plasma-workspace-x11`, disabled sddm `systemctl disable sddm`, and rebooted to `aurora:stable 41.20241208.4`, logged in as my user and issued `systemctl start lightdm.service`, and _lightdm's greeter_ showed up and worked (Later I did `systemctl enable lightdm.service` to boot directly to _lightdm_).
From lightdm I logged in to a Plasma-wayland session, and plasmashell crashed as described before by @renner0e when issuing `startplasma-wayland` at the commandline.
Then, after rebooting, tried a Plasma-x11 session, and it crashed as well.
Rebooted again and I logged in to a Gnome-wayland session, and the desktop loaded, barebones but functional. (I'm in fact posting this from the Gnome session)
I also tried using _gdm_ instead of _lightdm_, repeating the previous sequence, with similar results.
On the Gnome session I tried running different apps, and found that GTK apps run well, and so do Flatpak apps (some of them anyway. More on that later). Also apps in a Debian distrobox run well. QT apps, on the other hand, tend to crash, though some of them load. Here are some examples, starting them from the commandline:
❯ kate
 ~ took 2s
Kate doesn't show its window, but it shows in the activities view.
❯ ark
KCrash: Application 'ark' crashing... crashRecursionCounter = 2
Instrucción ilegal ('core' generado)
 ~ took 13s 
Ark plainly crashes.
❯ dolphin 
kf.config.core: Watching absolute paths is not supported "/usr/share/color-schemes
BreezeLight.colors"
KCrash: Application 'dolphin' crashing... crashRecursionCounter = 2
Instrucción ilegal ('core' generado)
 ~ took 15s 
Dolphin crashed the first day, but lately it shows its window but keeps sending the _KCrash: Application 'dolphin' crashing... crashRecursionCounter = 2_ message to the terminal, and after a while the system becomes completely non-responsive.
❯ konsole
kf.config.core: Watching absolute paths is not supported "/usr/share/color-schemes
BreezeLight.colors"
 ~ took 1m46s
Konsole used to show its window, and seemed to work fine, but lately is hanging the system similarly to Dolphin.
Okular, which is a flatpak, shows content correctly, and all seems to work, except the _open_ and _save_ dialogs,  which I suppose are loaded from the base Aurora system.

So, summarizing:
* sddm is crashing silently
* plasma-wayland and plasma-x11 both crash
* GTK apps run well (on the Gnome session)
* QT apps tend to crash or to hang the system.

Based on that, I would say that the issue is affecting some QT library/ies and that's what brings the system down.
I wonder if that "Instrucción ilegal ('core' generado)" message means that we can get some additional info on the issue. If so, please point me out to how to get it so I can post it here.

**Comment by renner0e** [comment]
I think I have finally made some progress into this matter:

the package `kf6-kimageformats` should be responsible for the breakage on this specific hardware.

I don't know why it not in the output of `rpm-ostree db diff`.

This affects aurora/bazzite because it is added in kinoite-main by ublue.
See https://github.com/ublue-os/main/blob/main/packages.json#L129


It is not present in upstream fedora, thus vanilla fedora works.

# **Workaround till a fix (probably upstream) is found.**
`sudo rpm-ostree override remove kf6-kimageformats` and rebooted
on an up-to-date `aurora:latest` and it finally works. 

```
● ostree-unverified-registry:ghcr.io/ublue-os/aurora:latest
Digest: sha256:00dcd0b8d0b00fa785b87395e4036bd1330f5ec2e4b955fe257b6f0b6bc6fea1
Version: latest-41.20241216 (2024-12-16T09:28:59Z)
RemovedBasePackages: kf6-kimageformats 6.8.0-1.fc41
```
## More Info
```
Name        : kf6-kimageformats
Version     : 6.8.0
Release     : 1.fc41
Architecture: x86_64
Install Date: Mi 11 Dez 2024 05:15:15 CET
Group       : Unspecified
Size        : 1210693
License     : LGPLv2+
Signature   : RSA/SHA256, Do 07 Nov 2024 17:06:12 CET, Key ID d0622462e99d6ad1
Source RPM  : kf6-kimageformats-6.8.0-1.fc41.src.rpm
Build Date  : Do 07 Nov 2024 14:20:20 CET
Build Host  : buildvm-x86-11.iad2.fedoraproject.org
Packager    : Fedora Project
Vendor      : Fedora Project
URL         : https://invent.kde.org/frameworks/kimageformats
Bug URL     : https://bugz.fedoraproject.org/kf6-kimageformats
Summary     : KDE Frameworks 6 Tier 1 addon with additional image plugins for QtGui
Description :
This framework provides additional image format plugins for QtGui.  As
such it is not required for the compilation of any other software, but
may be a runtime requirement for Qt-based software to support certain
image formats.
```

It is getting late for me.
I will link an upstream bug report in the coming days and report it on either fedora's or kde's bugzilla if it doesn't exist yet and further look into the issue.

# TODO:
https://community.kde.org/Guidelines_and_HOWTOs/Debugging/How_to_create_useful_crash_reports

figure out which debug packages have to be layered to create useful info to kde devs

enable fedora-debuginfo repo in `/etc/yum.repos.d/fedora.repo`

**Comment by renner0e** [comment]
`libheif` and `heif-pixbuf-loader` from negativo seem to cause the issue.

test from a "broken" image (with `kf6-kimageformats` installed)
## Workaround
`sudo rpm-ostree override replace --experimental --from repo="fedora" libheif heif-pixbuf-loader`

The packages from regular fedora repos work

```
● ostree-unverified-registry:ghcr.io/ublue-os/aurora:latest
Digest: sha256:de7067e0df0dfcc840ceb185cbc8fcc8215396bdec83b4cc1fd18d001c1bc9c1
Version: latest-41.20241218.1 (2024-12-18T07:55:07Z)
RemoteOverrides: repo=fedora
├─ heif-pixbuf-loader 1:1.18.2-5.fc41 -> 1.17.6-2.fc41
└─ libheif 1:1.18.2-5.fc41 -> 1.17.6-2.fc41

Downgraded:
heif-pixbuf-loader 1:1.18.2-5.fc41 -> 1.17.6-2.fc41
libheif 1:1.18.2-5.fc41 -> 1.17.6-2.fc41
```

vanilla fedora ships 1.17.6
```
Name        : libheif
Version     : 1.17.6
Release     : 2.fc41
Architecture: x86_64
Install Date: Mi 18 Dez 2024 17:26:51 CET
Group       : Unspecified
Size        : 966069
License     : LGPL-3.0-or-later and MIT
Signature   : RSA/SHA256, Fr 19 Jul 2024 06:59:09 CEST, Key ID d0622462e99d6ad1
Source RPM  : libheif-1.17.6-2.fc41.src.rpm
Build Date  : Fr 19 Jul 2024 01:35:35 CEST
Build Host  : buildvm-x86-27.iad2.fedoraproject.org
Packager    : Fedora Project
Vendor      : Fedora Project
URL         : https://github.com/strukturag/libheif
Bug URL     : https://bugz.fedoraproject.org/libheif
Summary     : HEIF and AVIF file format decoder and encoder
Description :
libheif is an ISO/IEC 23008-12:2017 HEIF and AVIF (AV1 Image File Format)
file format decoder and encoder.
```
and negativo ships 1.18.2
```
Name        : libheif
Epoch       : 1
Version     : 1.18.2
Release     : 5.fc41
Architecture: x86_64
Install Date: Mi 11 Dez 2024 05:14:59 CET
Group       : Unspecified
Size        : 1614755
License     : LGPLv3+ and MIT
Signature   : RSA/SHA256, Do 05 Dez 2024 00:49:44 CET, Key ID 97f3008993e8909b
Source RPM  : libheif-1.18.2-5.fc41.src.rpm
Build Date  : Mi 04 Dez 2024 23:42:23 CET
Build Host  : wks2.lab.negativo17.org
URL         : https://github.com/strukturag/libheif
Summary     : ISO/IEC 23008-12:2017 HEIF and AVIF file format decoder and encoder
Description :
libheif is an ISO/IEC 23008-12:2017 HEIF and AVIF (AV1 Image File Format) file
format decoder and encoder.

HEIF and AVIF are new image file formats employing HEVC (h.265) or AV1 image
coding, respectively, for the best compression ratios currently possible.

```

This is likely the commit that caused it:
https://github.com/negativo17/libheif/commit/4a4cbd901fa7c4209869a724976ef20a6796ff9e
Also coincides with the date as this was committed on Nov 12th, after that date the images broke.
```
%changelog
* Tue Nov 12 2024 Simone Caronni <negativo17@gmail.com> - 1:1.18.2-4
- Re-enable OpenJPH.
```

The package was actually in the diff @mejiasjrg sent earlier from Nov 12th ->13th. I don't know how I missed that. It is right at the top.

## Possibilities:
- the library is broken (it should be broken on GNOME as well if this is the case, but nobody happened to notice as everything is flatpak anway)
- some system QT component acting weirdly with that additional codec support enabled for whatever reason
- regression in 1.18

Will test on either Arch Linux or KDE Linux to confirm the `libheif` issue.

On kde-linux (not kde-neon, the new thing) they ship the new 1.19 version and the live environment works.
https://gitlab.archlinux.org/archlinux/packaging/packages/libheif/-/blob/main/PKGBUILD?ref_type=heads

**Comment by mejiasjrg** [comment]
> I think I have finally made some progress into this matter:
> 
> the package `kf6-kimageformats` should be responsible for the breakage on this specific hardware.
> 
> If you layer this package on vanilla fedora, everything still works. If you remove it from kinoite-main and aurora, it works again.
> 
> I don't know why it not in the output of `rpm-ostree db diff`.
> 
> This affects aurora/bazzite because it is added in kinoite-main by ublue. See https://github.com/ublue-os/main/blob/main/packages.json#L129
> 
> It is not present in upstream fedora, thus vanilla fedora works.
> # **Workaround till a fix (probably upstream) is found.**
> 
> `sudo rpm-ostree override remove kf6-kimageformats` and rebooted on an up-to-date `aurora:latest` and it finally works.
> 

Very well done @renner0e !!!
My machine is up and running _plasma_ on `aurora:stable` after applying your fix (`sudo rpm-ostree override remove kf6-kimageformats`):
❯ rpm-ostree status
State: idle
AutomaticUpdates: stage; rpm-ostreed-automatic.timer: no runs since boot
Deployments:
● ostree-image-signed:docker://ghcr.io/ublue-os/aurora:stable
                   Digest: sha256:c71db2eea4e9c5f6715fda2560f15a153f4b7f86d15c2fa490d8b538bad189d9
                  Version: 41.20241216.1 (2024-12-16T13:41:20Z)
      RemovedBasePackages: kf6-kimageformats 6.8.0-1.fc41
          LayeredPackages: gdm lightdm lightdm-greeter lightdm-settings plasma-workspace-x11

  ostree-image-signed:docker://ghcr.io/ublue-os/aurora:stable
                   Digest: sha256:c71db2eea4e9c5f6715fda2560f15a153f4b7f86d15c2fa490d8b538bad189d9
                  Version: 41.20241216.1 (2024-12-16T13:41:20Z)
          LayeredPackages: gdm lightdm lightdm-greeter lightdm-settings plasma-workspace-x11

Of course, I still have to clean up the mess I made (layered packages), but that will be tomorrow.
Thank you!

**Comment by ledif** [comment]
Duplicate of #7 and blocked by https://github.com/ublue-os/main/issues/695.

**Comment by ledif** [comment]
This is now fixed with #153.
